SalesforceAnalytics.podspec

```diff
@@ -14,6 +14,8 @@ Pod::Spec.new do |s|
                      :tag => "v#{s.version}",
                      :submodules => true }
 
+  s.frameworks   = 'CoreTelephony'
```

Comment:
CoreTelephony is supported in Catalyst but needs to be explicitly linked to build successfully (for example, on core we didn’t and that worked for everything else but not Catalyst).   

So far I just added CoreTelephony to the frameworks here but if we go with that route I’m thinking we should include the other frameworks (UIKit, Fundation) to be consistent. The other option is to skip the CoreTelephony related code for Catalyst, usages [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/9f4daa6f28fb1dbeaeee0e611bd550ece66e408e/libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Model/SFSDKInstrumentationEventBuilder.m#L109) and [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/9f4daa6f28fb1dbeaeee0e611bd550ece66e408e/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/UIDevice%2BSFHardware.m#L640). Any preferences?

Reply:
Let's add it in for now - we need those methods.